editors/code/src/client.ts

```diff
@@ -1,9 +1,33 @@
 import * as lc from 'vscode-languageclient';
 import * as vscode from 'vscode';
+import * as lcx from './lsp_ext';
 
 import { CallHierarchyFeature } from 'vscode-languageclient/lib/callHierarchy.proposed';
 import { SemanticTokensFeature, DocumentSemanticsTokensSignature } from 'vscode-languageclient/lib/semanticTokens.proposed';
 
+function toTrusted(obj: vscode.MarkedString): vscode.MarkedString {
+    const md = <vscode.MarkdownString>obj;
+    if (md && md.value.includes("```rust")) {
+        md.isTrusted = true;
+        return md;
+    }
+    return obj;
+}
```

Comment:
Is this really trusted? Should we do some kind of validation/escaping here?

Reply:
Good catch!

Indeed someone can write in a doc comment something like  
`[title]('command:some.vscode.command')`  
and we will render the corresponding link in the hover without any validation. I left this code in the hope it would be useful to implement intra-doc links later. But now I think I'd better remove it, and if we will support intra-doc links we have to add some validation. But not in this PR.

Command links added to the hover bottom are safe as we fully control them. So this code is  safe in turn: [ hover.contents.push(renderHoverActions(actions)); ](https://github.com/vsrs/rust-analyzer/blob/a0dfb72edbabb22935d05c2b7f73c9ae72fc9284/editors/code/src/client.ts#L70)